scc/src/opts.rs

```diff
@@ -49,86 +49,177 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
-    let value = positional::<PathBuf>("SCRIPT_PATH");
-    construct!(tag, value).anywhere().map(|pair| pair.1).many().catch()
+fn scripts_dir() -> impl Parser<Option<PathBuf>> {
+    let tag = any::<String>("-compile")
+        .help("Compile scripts to blob")
+        .guard(|s| s == "-compile", "not compile");
+    let value = positional::<PathBuf>("SCRIPT_PATH").guard(SlongArg::is_not_slong, "starts with -");
+    construct!(tag, value).adjacent().map(|pair| pair.1).optional().catch()
 }
 
 fn script_paths_file() -> impl Parser<Option<PathBuf>> {
     let tag = any::<String>("-compilePathsFile")
-        .help("File listing all of the script paths")
+        .help("File containing a newline-delimited list of redscript paths to compile")
         .guard(|s| s == "-compilePathsFile", "not compilePathsFile");
-    let value = positional::<PathBuf>("FILE");
-    construct!(tag, value).anywhere().map(|pair| pair.1).optional().catch()
+    let value = positional::<PathBuf>("SCRIPT_PATHS_FILE").guard(SlongArg::is_not_slong, "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 fn cache_dir() -> impl Parser<Option<PathBuf>> {
-    let tag = any::<String>("-customCacheDir").guard(|s| s == "-customCacheDir", "not customCacheDir");
-    let value = positional::<PathBuf>("CACHE_DIR");
-    construct!(tag, value).map(|pair| pair.1).optional().catch()
+    let tag = any::<String>("-customCacheDir")
+        .help("A custom cache dir to write the final.redscripts to")
+        .guard(|s| s == "-customCacheDir", "not customCacheDir");
+    let value = positional::<PathBuf>("CACHE_DIR").guard(SlongArg::is_not_slong, "starts with -");
+    construct!(tag, value)
+        .adjacent()
+        .anywhere()
+        .map(|pair| pair.1)
+        .optional()
+        .catch()
 }
 
 impl Opts {
-    pub fn load(args: &[&str]) -> Self {
-        let optimize = toggle_options("optimize", "Optimize the redscripts");
-        let threads = slong("-threads", "Number of theads to script_paths on")
-            .parse(|s| s.parse::<u8>())
-            .fallback(1);
-        let no_warnings = toggle_options("Wnone", "No warnings");
-        let no_testonly = toggle_options("no-testonly", "No testonly classes");
-        let no_breakpoint = toggle_options("no-breakpoint", "No breakpoints");
-        let profile_off = toggle_options("profile=off", "Profile off");
-        let cache_file = any::<PathBuf>("CACHE_FILE").optional();
+    pub fn get_parser() -> OptionParser<Opts> {
+        let optimize = toggle_options("-optimize", "Enable optimiziations. Off by default")
+            .map(|s| s.unwrap_or(Opts::default().optimize));
+        let threads = slong("-threads", "THREADS", "Set number of internal compilation threads")
+            .parse(|s| s.map(|s| s.parse::<u8>()).unwrap_or(Ok(Opts::default().threads)));
```

Comment:
what I meant about using `const` was not making the `default` method const, I just meant extracting each field to it's own `const` like this:
```rs
const DEFAULT_THREADS: usize = 1;
```
and then referring to the consts directly:
```suggestion
            .parse(|s| s.map(|s| s.parse::<u8>()).unwrap_or(Ok(DEFAULT_THREADS)));
```

Reply:
Isn't `const fn` evaluated at compile time too? I thought it'd have nearly the same performance, and editing all the defaults in one place (with their types already defined) seemed a bit cleaner.

Reply:
it is evaluated at compile time, and it'll probably have similar performance, but this just seems a little roundabout to me, we don't really need the `default` method, we just need defaults for some of the options inside

this could also lead to complications if the default instance performs some side effects or allocations or something else that can't be made `const`, so I'd rather stick to just having a few private `const DEFAULT_{x}` items